From 5bb23a6bf8d7b5bd3c2dacb88a7b140b37cfc861 Mon Sep 17 00:00:00 2001
From: Atul Dhudase <quic_adhudase@quicinc.com>
Date: Fri, 8 Mar 2024 13:03:23 +0530
Subject: [PATCH] qdl: add logs indicating flashing progress

Add logs in qdl to indicate file being flashed and
show flashing progress.
---
 firehose.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/firehose.c b/firehose.c
index da3ed31..8e95f49 100644
--- a/firehose.c
+++ b/firehose.c
@@ -307,6 +307,7 @@ static int firehose_erase(struct qdl_device *qdl, struct program *program)
 	xml_setpropf(node, "num_partition_sectors", "%d", program->num_sectors);
 	xml_setpropf(node, "start_sector", "%s", program->start_sector);
 
+	fprintf(stderr, "[PROGRAM] Please wait, flashing \"%s\" now...\n", program->label);
 	ret = firehose_write(qdl, doc);
 	if (ret < 0) {
 		fprintf(stderr, "[PROGRAM] failed to write program command\n");
@@ -391,6 +392,7 @@ static int firehose_program(struct qdl_device *qdl, struct program *program, int
 	lseek(fd, (off_t) program->file_offset * program->sector_size, SEEK_SET);
 	left = num_sectors;
 	while (left > 0) {
+		fprintf(stderr, "[PROGRAM] %d sectors remaining out of %d\r", left, num_sectors);
 		chunk_size = MIN(max_payload_size / program->sector_size, left);
 
 		n = read(fd, buf, chunk_size * program->sector_size);
@@ -410,6 +412,8 @@ static int firehose_program(struct qdl_device *qdl, struct program *program, int
 		left -= chunk_size;
 	}
 
+	fprintf(stderr, "\n");
+
 	t = time(NULL) - t0;
 
 	ret = firehose_read(qdl, 30000, firehose_generic_parser, NULL);
-- 
2.34.1

